src/components/Search/SearchBox.tsx

```diff
@@ -0,0 +1,150 @@
+        !activeRoutesQuery.isLoading && !activeRoutesQuery.isError && activeRoutesQuery.data !== undefined,
+    }
+  );
+
+  const archivedRouteMatcherQuery = useQuery(
+    [SearchView.ArchivedRoutes, 'matcher'],
+    async () => getArchivedRoutesSubstringMatcher(),
+    {
+      enabled: view === SearchView.ArchivedRoutes,
+    }
+  );
+
+  // reset results for when our view prop changes
+  useEffect(() => {
+    setResults([]);
+  }, [view]);
+
+  const updateSearchResults = useCallback(async () => {
+    if (view === SearchView.Users) {
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      const userSearchResults: UserSearchResult[] = userMatcherQuery.data!.getMatches(inputText);
+      const users: User[] = userSearchResults.map((userSearchResult: UserSearchResult) => userSearchResult.user);
+
+      setResults(users.map((currUser: User) =>
+        <VStack key={currUser.getId()} width='100%'>
+          <Divider orientation='horizontal' />
+          <UserRow user={currUser} onPress={onSelect} />
+        </VStack>
+      ));
+    }
+    else if (view == SearchView.ActiveRoutes) {
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      const activeRouteSearchResults: string[] = activeRouteMatcherQuery.data!.getMatches(inputText);
+      const activeRoutes: Route[] = await Promise.all(activeRouteSearchResults.map(getRouteByName));
+
+      setResults(activeRoutes.map((currRoute: Route) =>
+        <VStack key={currRoute.getId()} width='100%'>
+          <Divider orientation='horizontal' />
+          <RouteRow route={currRoute} onPress={onSelect} />
+        </VStack>
+      ));
+    }
+    else if (view === SearchView.ArchivedRoutes) {
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      const archivedRouteSearchResults: string[] = archivedRouteMatcherQuery.data!.getMatches(inputText);
+      const archivedRoutes: Route[] = await Promise.all(archivedRouteSearchResults.map(getRouteByName));
+
+      setResults(archivedRoutes.map((currRoute: Route) =>
+        <VStack key={currRoute.getId()} width='100%'>
+          <Divider orientation='horizontal' />
+          <RouteRow route={currRoute} onPress={onSelect} />
+        </VStack>
+      ));
+    }
+  }, [view, userMatcherQuery, activeRouteMatcherQuery, archivedRouteMatcherQuery, inputText]);
+
+  // get which matcher query we are referring to for checking for loading and errors
+  const matcherQuery = (
+    view === SearchView.Users ? userMatcherQuery :
+      (view === SearchView.ActiveRoutes ? activeRouteMatcherQuery :
+        archivedRouteMatcherQuery));
+
+  if (matcherQuery.isLoading) {
+    return null;
+  }
+
+  if (matcherQuery.isError || matcherQuery.data === undefined) {
+    console.error(matcherQuery.error);
+    return null;
+  }
+
+  return (
+    <Box width={width} maxH={maxHeight ?? 'auto'} paddingRight={maxHeight !== undefined ? '1' : '0'}>
```

Comment:
Besides the `useCallback` change, the only other thing I see is that the maxHeight should be passed to the `ScrollView` itself rather than the `Box` because what we want to make scrollable are the results itself, rather than the results + the `Input` box